cirq/google/engine/engine.py

```diff
@@ -65,7 +65,7 @@ class ProtoVersion(enum.Enum):
 def _make_random_id(prefix: str, length: int = 16):
     random_digits = [random.choice(string.ascii_uppercase + string.digits) for _ in range(length)]
     suffix = ''.join(random_digits)
-    suffix += datetime.date.today().strftime('%y%m%d')
+    suffix += datetime.date.today().strftime('%y%m%d-%H%M%S')
```

Comment:
why not use UUID?

Reply:
The problem with this second based approach that it assumes that if the faulty (assumingly seeding) client program is generating more than one program per second, you'll still get collisions. With UUID, neither seeding nor speed won't be a problem - though it will be longer (36 characters).

Reply:
A UUID based on what?  I am not sure if that solves the problem. 
Would you hash based on the time + circuit?  If we use a random uuid, then that would be based on random() and also fail if seeded.

Reply:
+1